spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/CassandraRunner.scala

```diff
@@ -67,14 +65,21 @@ private[connector] class CassandraRunner(val configTemplate: String, props: Map[
       logConfigFileProperty, "-cp", classPath, cassandraMainClass, "-f")
     .inheritIO()
     .start()
+
   val startupTime = System.currentTimeMillis()
 
-  val nativePort = props.get("native_transport_port").get.toInt
-  if (!waitForPortOpen(InetAddress.getByName(props.get("rpc_address").get), nativePort, 100000))
-    throw new IOException("Failed to start Cassandra.")
+  if (!waitForPortOpen(InetAddress.getByName(baseConfiguration.rpcAddress), baseConfiguration.nativeTransportPort, 100000, () => !process.isAlive)) {
```

Comment:
because Cassandra may be starting but it may not start. Say we started the process, Cassandra needs some time to startup, we passed the condition "if process is alive", and we are waiting for 100 seconds to acquire the connection. But Cassandra could die in the meantime, so there is no point in waiting any longer.